### ISSUE TITLE
Add enum constructor to create enum from value 

### ISSUE THREAD

**Issue by aisk** [Thriftpy/thriftpy2#111]
Thought we have a thrift file:

```thrift
enum TestEnum {
    a = 1;
    b = 2;
}
```

It's common to create a enum from it's value like this:

```python
a = TestEnum(1)
```

For now, the enum type do not accept any constructor parameters, can we add it like above, or generated a public api to do this convert?

**Comment by ethe** [comment]
Yes, I think it is better to have the same behavior to https://docs.python.org/3/library/enum.html

**Comment by seidefrank** [comment]
Is there a workaround? I need to use an API that, per `.pyi` file, expects an enum type, but I am only given an integer value. Is it possible to convert the `int` value to an instance of the Thrift-generated enum type? Thanks!

**Comment by truebit** [comment]
+1
Now thriftpy2 just treats thrift enum type to TType.I32. 
I think we need to add a new TType ENUM.
